policy/modules/admin/tomoyo-auditd.te

```diff
@@ -0,0 +1,32 @@
+policy_module(tomoyo-auditd, 1.0)
+
+########################################
+#
+# Declarations
+#
+
+type tomoyo_auditd_t;
+type tomoyo_auditd_exec_t;
+init_daemon_domain(tomoyo_auditd_t, tomoyo_auditd_exec_t)
+
+type tomoyo_auditd_etc_t;
+files_config_file(tomoyo_auditd_etc_t)
+
+type tomoyo_auditd_var_log_t;
+logging_log_file(tomoyo_auditd_var_log_t)
+
+########################################
+#
+# Tomoyo-auditd local policy
+#
+
+dev_rw_null(tomoyo_auditd_t)
+
+files_read_etc_files(tomoyo_auditd_t)
+read_files_pattern(tomoyo_auditd_t, tomoyo_auditd_etc_t, tomoyo_auditd_etc_t)
+
+logging_log_filetrans(tomoyo_auditd_t, tomoyo_auditd_var_log_t, file)
+manage_files_pattern(tomoyo_auditd_t, tomoyo_auditd_var_log_t, tomoyo_auditd_var_log_t)
```

Comment:
you might be able to enhance security by omitting the "file write" here. log files are generally opened for append only.
by ommitting the write you enhance the integrity of your audit trail because a compromised tomoyo-auditd is not able to remove entries from the log (only append to it)

Reply:
Yes, the daemon only uses `O_APPEND`.